### PR TITLE
R and rpy2

### PR DESCRIPTION
**Pull request by sankhesh**
Enable R and RPY2 support.

@doutriaux1 Kindly review

### CODE REVIEW

**Comment by sankhesh** [comment]
@aashish24

File: CMake/cdat_modules/r_external.cmake [hunk 1]
```diff
@@ -16,7 +21,8 @@ ExternalProject_Add(R
   URL_MD5 ${R_MD5}
   BUILD_IN_SOURCE 1
   PATCH_COMMAND ""
-  CONFIGURE_COMMAND <SOURCE_DIR>/configure --prefix=<INSTALL_DIR> LIBnn=lib --without-jpeglib --disable-R-framework --enable-R-shlib --disable-openmp --without-cairo --without-ICU --without-libpng --without-system-xz --without-aqua --without-tcltk --without-readline --with-x=${WITHX}
+  DEPENDS ${R_deps}
+  CONFIGURE_COMMAND env ${USR_ENVS} <SOURCE_DIR>/configure --prefix=<INSTALL_DIR> LIBnn=lib --without-jpeglib --disable-R-framework --enable-R-shlib --disable-openmp --without-cairo --without-ICU --without-libpng --without-system-xz --without-aqua --without-tcltk --with-x=${WITHX}
```
Comment (doutriaux1) [inline CMake/cdat_modules/r_external.cmake]: @sankhesh could you try to build uvcdat with -DCDAT_BUILD_PARALLEL=ON and turn on openmp in here?

Thanks.

File: CMake/cdat_modules/r_pkg.cmake [hunk 2]
```diff
@@ -1,9 +1,9 @@
 set(R_MAJOR_SRC 3)
-set(R_MINOR_SRC 1)
-set(R_PATCH_SRC 0)
-set(R_URL ${LLNL_URL})
+set(R_MINOR_SRC 2)
+set(R_PATCH_SRC 1)
+set(R_URL "http://cran.r-project.org/src/base/R-3")
```
Comment (doutriaux1) [inline CMake/cdat_modules/r_pkg.cmake]: ok I will tweak this back to LLNL when it works, it's "safer" to leave the links pointing to our server, less likely to break in the future.

File: CMake/cdat_modules/rpy2_external.cmake [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# create an external project to install RPY2,
+# and configure and build it
+set(nm RPY2)
+
+# Set PATH and R_HOME to find R
+list(APPEND USR_ENVS
+  "R_HOME=${cdat_EXTERNALS}/lib/R"
```
Comment (doutriaux1) [inline CMake/cdat_modules/rpy2_external.cmake]: @aashish24 when we enable the options to use system's externals, let's make sure to set this correctly.
@sankhesh could you set this only if 
1- CDAT_BUILD_R is ON and R_HOME is not already set?

File: CMake/cdat_modules/r_external.cmake (hunk 1 above)
Comment (doutriaux1) [inline CMake/cdat_modules/r_external.cmake]: @sankhesh we should check if user has R_HOME set at this point, is so we should error exit and tell user to turn either unset it or not build R from here.

Comment (aashish24) [inline CMake/cdat_modules/r_external.cmake]: @doutriaux1 let's do then when we work on the system package overhaul.

File: CMake/cdat_modules/rpy2_external.cmake (hunk 3 above)
Comment (aashish24) [inline CMake/cdat_modules/rpy2_external.cmake]: @doutriaux1 CDAT_BUILD_R will be ON if rpy2 is building since we are not supporting system R.

Comment (doutriaux1) [inline CMake/cdat_modules/rpy2_external.cmake]: ok let's wait until we allow for systemr to address this.

**Comment by doutriaux1** [comment]
testring on my mac and ubuntu and merging.

**Comment by doutriaux1** [comment]
should we add a basic test "import e2py" to make sure the build went properly?

**Comment by sankhesh** [comment]
@doutriaux1 Yes. I'll be adding one soon.

**Comment by sankhesh** [comment]
Replaced with #1440

**Pull request closed without merging**